tests/kernel/timer/timer_behavior/prj.conf

```diff
@@ -1,3 +1,4 @@
 CONFIG_ZTEST=y
 CONFIG_ZTEST_NEW_API=y
 CONFIG_CBPRINTF_FP_SUPPORT=y
+CONFIG_FPU=y
```

Comment:
This shouldn't be needed, without the FPU there should be software generated fallbacks for float math from the compiler.

Reply:
After discussing this with others this seems like a bug on a particular platform. CONFIG_FPU=y should be selected by the board/soc probably if CONFIG_HAS_FPU=y as a default *or* the build flags should be set to infer CONFIG_FPU=n to mean "-msoft-float".

This flag isn't required on arm for example where a FPU may/may not exist and doesn't seem to cause fault probably because the implication is soft-float usage.

Reply:
thanks, there is indeed a soft float support for my x86 platform via config X86_FP_USE_SOFT_FLOAT, but I am not sure why this config relies on NEWLIB_LIBC, from the commit message in  ff720cd9b34, NEWLIB_LIBC should rely on X86_FP_USE_SOFT_FLOAT or FPU, so I updated the Kconfig dependency for them.

Then if without FPU, this soft float will be enabled in default for x86, we don't want to set FPU in default because this will incur performance penalty to save FPU registers for all threads during context switch.